src/main/java/org/sagebionetworks/bridge/reporter/worker/RetentionReportGenerator.java

```diff
@@ -65,20 +74,27 @@ public Report generate(BridgeReporterRequest request, String studyId) throws IOE
                         break;
                     }
                 }
+                
                 if (studyStartDate == null) {
                     LOG.error("No study_state_date event for id=" + accountSummary.getId());
                     continue;
                 }
-                RequestInfo requestInfo = bridgeHelper.getRequestInfoForParticipant(studyId, accountSummary.getId());
+                
+                RequestInfo requestInfo = bridgeHelper.getRequestInfoForParticipant(
+                        studyId, accountSummary.getId());
                 if (requestInfo.getSignedInOn() != null) {
-                    int sign_in_days = Days.daysBetween(studyStartDate.toLocalDate(), requestInfo.getSignedInOn().toLocalDate()).getDays();
+                    int sign_in_days = Days.daysBetween(studyStartDate, 
+                            requestInfo.getSignedInOn().withZone(DateTimeZone.UTC)).getDays();
+                    
                     while (signInData.size() < (sign_in_days + 1)) {
```

Comment:
Just to be extra safe, we should check if sign_in_days is somehow negative, and then continue on to the next participant if it is. (Same with upload_on_days.)